blueocean-jira/src/main/java/io/jenkins/blueocean/service/embedded/jira/JiraSCMListener.java

```diff
@@ -40,38 +47,53 @@ public void onChangeLogParsed(Run<?, ?> run, SCM scm, TaskListener listener, Cha
             return;
         }
         // Query for JIRA issues
-        Set<JiraIssue> issuesFromJqlSearch = Sets.newHashSet(Iterables.transform(session.getIssuesFromJqlSearch(jql), new Function<Issue, JiraIssue>() {
-            @Override
-            public JiraIssue apply(Issue input) {
-                return new JiraIssue(input);
+        try {
+            List<Issue> issues = session.getIssuesFromJqlSearch(jql);
+            Set<JiraIssue> issuesFromJqlSearch = issues == null ? Collections.emptySet() :
+                issues.stream().map( input -> new JiraIssue(input) )
+                    .collect( Collectors.toSet() );
+
+            // If there are no JIRA issues, do not update the actions
+            if (issuesFromJqlSearch.isEmpty()) {
+                return;
             }
-        }));
-        // If there are no JIRA issues, do not update the actions
-        if (issuesFromJqlSearch.isEmpty()) {
-            return;
-        }
-        // Create or update the JiraBuildAction
-        JiraBuildAction action = run.getAction(JiraBuildAction.class);
-        if (action == null) {
-            run.addAction(new JiraBuildAction(run, issuesFromJqlSearch));
-        } else {
-            action.addIssues(issuesFromJqlSearch);
+            // Create or update the JiraBuildAction
+            JiraBuildAction action = run.getAction(JiraBuildAction.class);
+            if (action == null) {
+                run.addAction(new JiraBuildAction(run, issuesFromJqlSearch));
+            } else {
+                action.addIssues(issuesFromJqlSearch);
+            }
+            run.save();
+        } catch (Exception e ){ // we do not want to fail the build if an issue happen here
+            LOGGER.warn( "skip reccording associated jira issues: {}", e.getMessage() );
```

Comment:
Typo, `s/reccording/recording`.  It should also say why its being skipped, that is in case `e.getMessage()` is not self explanatory? Maybe something like: `Failure executing Jira query to fetch issues. Skipping recording Jira issues.: {}`.

Reply:
done